core/workspace_svg.js

```diff
@@ -1797,12 +1806,34 @@ Blockly.WorkspaceSvg.prototype.setResizesEnabled = function(enabled) {
 };
 
 /**
+ * Set whether this workspace is currently doing a bulk update.
+ * A bulk update pauses workspace resizing but also pauses other expensive
+ * operations, such as refreshing the toolbox as variables are added and
+ * removed.
+ * @param {boolean} enabled True if a bulk update is starting, false if a bulk
+ *     update is ending.
+ * @package
+ */
+Blockly.WorkspaceSvg.prototype.setBulkUpdate = function(enabled) {
+  // This will trigger a resize if necessary.
+  this.setResizesEnabled(enabled);
+  var stoppedUpdating = (this.isBulkUpdating_ && !enabled);
+  this.isBulkUpdating_ = stoppedUpdating;
```

Comment:
I don't think this will work correctly. if you call `setBulkUpdate(true)` when `isBulkUpdating_` is false, `stoppedUpdating` will be false, so `isBulkUpdating` will never be set to true. Seems like it should be set to `enabled` instead?

Reply:
Good catch.